src/definition.ts

```diff
@@ -135,18 +136,9 @@ class API {
           content: raw,
         });
       }
-      absPath = paths.shift();
-    }
-
-    if (typeof mainReference === 'undefined') {
```

Comment:
Why can we remove this? We are now sure that the main definition can be returned, even if parsing doesn't work?

Reply:
Because we now initialize the variable with an “empty” state: 
`let mainReference: JSONSchemaWithRaw = { parsed: {}, raw: '' };`

So the case where we can't find the main reference will be handled by the error of line 144:

```typescript
    if (!parsed || !(parsed instanceof Object) || !('info' in parsed)) {
      throw new UnsupportedFormat(
        "Definition needs to be an object with at least an 'info' key",
      );
    }
```